lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -23,10 +23,24 @@ sub list {
     my $self = shift;
 
     my %args;
-    for my $arg (qw/state build iso distri version flavor maxage scope group limit/) {
+    for my $arg (qw/build iso distri version flavor maxage scope group limit/) {
         next unless defined $self->param($arg);
         $args{$arg} = $self->param($arg);
     }
+    # For these, we accept a single value, a single instance of the arg
+    # with a comma-separated list of values, or multiple instances of the
+    # arg. In all cases we're going to convert to an array ref of values:
+    # we could let query_jobs do the string splitting for us, but this is
+    # clearer.
+    for my $arg (qw/state ids/) {
+        next unless defined $self->param($arg);
+        if (index($self->param($arg), ',') != -1) {
+            $args{$arg} = [split(',', $self->param($arg))];
```

Comment:
my thought was that it's clearer if this function is known to *always* pass an array ref.